### PR TITLE
Create function for deep healthcheck route from S3

### PR DESCRIPTION
**Pull request by dora-korpar**
Creates route for deep healthcheck http request from S3

### CODE REVIEW

File: lib/IAMClient.js [hunk 1]
```diff
@@ -390,6 +390,12 @@ class IAMClient {
         }, data, options.reqUid, 'application/json', options.logger);
     }
 
+    healthcheck(log, callback) {
+        const logger = log || undefined;
```
Comment (dora-korpar) [inline lib/IAMClient.js]: yeah I meant to get rid of that. Good eye

Comment (electrachong) [inline lib/IAMClient.js]: okie

File: lib/IAMClient.js [hunk 2]
```diff
@@ -546,9 +542,10 @@ class IAMClient {
             }
             /* This is not an error */
             if (res.statusCode < 400) {
-                return callback(null, obj, res.statusCode);
+                return callback(null, obj, res.statusCode, res.statusMessage);
             }
             // Load the error from errors(arsenal)
+            console.log('in handle response', obj);
```
Comment (bennettbuchanan) [inline lib/IAMClient.js]: Looks like you left a `console.log` here. 😃

**Comment by dora-korpar** [comment]
@ironman-machine work some magic

**Comment by dora-korpar** [comment]
@ironman-machine get movin

File: tests/unit/healthcheckTest.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            client = new IAMClient('127.0.0.1', 8500);
+            done();
+        }).on('error', done);
+    });
+
+    afterEach('stop server', () => { server.close(); });
+
+    it('should return 200 on request', done => {
+        client.healthcheck(null, (err, obj, resCode) => {
+            assert.deepEqual(err, null);
```
Comment (electrachong) [inline tests/unit/healthcheckTest.js]: possible to use assert.deepStrictEqual instead?

File: tests/unit/healthcheckTest.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+            assert.deepEqual(resCode, 200);
```
Comment (electrachong) [inline tests/unit/healthcheckTest.js]: same here

File: tests/unit/healthcheckTest.js (hunk 3 above)
Comment (dora-korpar) [inline tests/unit/healthcheckTest.js]: yep!

File: tests/unit/healthcheckTest.js (hunk 4 above)
Comment (dora-korpar) [inline tests/unit/healthcheckTest.js]: yesiree

**Comment by dora-korpar** [comment]
@ironman-machine just to be sure

**Pull request merged**